src/mono/mono/metadata/handle.h

```diff
@@ -517,7 +517,10 @@ TYPED_HANDLE_DECL (MonoException);
 TYPED_HANDLE_DECL (MonoAppContext);
 
 /* Simpler version of MONO_HANDLE_NEW if the handle is not used */
-#define MONO_HANDLE_PIN(object) MONO_HANDLE_NEW (MonoObject, (MonoObject*)(object))
+#define MONO_HANDLE_PIN(object) do { \
+		if ((object) != NULL) \
```

Comment:
I don't think the null check is needed.

Reply:
Some object fields of the `ReflectionMethodBuilder` can be null. Allocating a handle in that case is pointless. This avoid doing explicit checks for each field.